File: llm_generated_code/Gemini/Python/Level2/Q12.py
def climb_stairs(n):
    """
    Calculates the number of ways to climb n stairs.

    Args:
        n: The number of stairs.

    Returns:
        The number of distinct ways to climb the stairs.
    """

    if n <= 1:
        return 1

    dp = [0] * (n + 1)
    dp[0] = 1
    dp[1] = 1

    for i in range(2, n + 1):
        dp[i] = dp[i - 1] + dp[i - 2]

    return dp[n]

# Example usage:
n = 4
result = climb_stairs(n)
print("Number of ways to climb:", result)  # Output: 5

File: llm_generated_code/Gemini/CPP/Level2/Q12.cpp
#include <iostream>

using namespace std;

int climbStairs(int n) {
    if (n <= 2) {
        return n;
    }

    int first = 1, second = 2;
    int third;

    for (int i = 3; i <= n; i++) {
        third = first + second;
        first = second;
        second = third;
    }

    return third;
}

int main() {
    int n = 4; // Number of stairs
    int ways = climbStairs(n);
    cout << "Number of ways to climb " << n << " stairs: " << ways << endl;

    return 0;
}
